internal/api/handlers/custom.go

```diff
@@ -0,0 +1,23 @@
+package handlers
+
+import (
+	"errors"
+	"net/http"
+)
+
+type HandlerFunc func(w http.ResponseWriter, r *http.Request) error
+
+// CustomHandler is a wrapper that allows us to process and return errors from different handlers.
+func CustomHandler(f HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		err := f(w, r)
+		if err != nil {
+			status := http.StatusInternalServerError
+			var apiError *Error
+			if errors.As(err, &apiError) {
+				status = apiError.HttpStatus()
+			}
```

Comment:
Is the status all you care about in the API error? Are there any extra fields you'd like to render differently?

Reply:
I added an extras field too, based on the SDP implementation